stellarphot/settings/tests/test_photometry_runner.py

```diff
@@ -0,0 +1,100 @@
+from stellarphot.settings.tests.test_models import DEFAULT_PHOTOMETRY_SETTINGS
+
+
+# See test_settings_file.TestSavedSettings for a detailed description of what the
+# following fixture does. In brief, it patches the settings_files.PlatformDirs class
+# so that the user_data_dir method returns the temporary directory.
+@pytest.fixture(autouse=True)
+def fake_settings_dir(mocker, tmp_path):
+    mocker.patch.object(
+        settings_files.PlatformDirs, "user_data_dir", tmp_path / "stellarphot"
+    )
+
+
+class TestPhotometryRunner:
+
+    # This auto-used fixture changes the working directory to the temporary directory
+    # and then changes back to the original directory after the test is done.
+    @pytest.fixture(autouse=True)
+    def change_to_tmp_dir(self, tmp_path):
+        original_dir = os.getcwd()
+        os.chdir(tmp_path)
+        # Yielding here is important. It means that when the test is done, the remainder
+        # of the function will be executed. This is important because the test is run in
+        # a temporary directory and we want to change back to the original directory
+        # when the test is done.
+        yield
+        os.chdir(original_dir)
+
+    def test_photometry_runner_creation(self):
+        # This test simply makes sure we can create the object
+        photometry_runner = PhotometryRunner()
+        assert isinstance(photometry_runner, ipw.Box)
+
+    @pytest.mark.parametrize("do_photometry", [True, False])
+    def test_photometry_runner_with_valid_settings(self, do_photometry):
+        # This test makes sure that the PhotometryRunner widget can be used
+        # to *start* a photometry run. It does not check the actual photometry
+        # or the results of the photometry run, except to ensure that the expected
+        # files are created.
+        # Make a settings
+        phot_settings = PhotometrySettings.model_validate(DEFAULT_PHOTOMETRY_SETTINGS)
+
+        # Save a copy to the working directory
+        wd_settings = PhotometryWorkingDirSettings()
+        wd_settings.save(phot_settings)
+
+        # Make sure a source list file exists
+        sources = Path(phot_settings.source_location_settings.source_list_file)
+        sources.touch()
+
+        # Make sure a fake fits file exists
+        fake_fits = Path("fake.fits")
+        fits_data = fits.PrimaryHDU(data=[[1, 2], [3, 4]])
+        object_name = "Fake Object"
+        fits_data.header["object"] = object_name
+        fits_data.writeto(fake_fits)
+
+        photometry_runner = PhotometryRunner()
+        # Select a fits file
+        photometry_runner.fo.file_chooser.reset(".", fake_fits.name)
+        photometry_runner.fo.file_chooser._apply_selection()
+        photometry_runner.fo.file_chooser.value = fake_fits
+
+        # Check that the message in the information box is as expected
+        assert "Photometry will be done" in photometry_runner.info_box.value
```

Comment:
Presumably if you listened to my earlier recommendation to replace "done" with "performed" then this line of the test will need updating.

Reply:
You were correct, sir. I didn't follow your advice initially, but you were correct...